locator/topology.cc

```diff
@@ -585,6 +591,19 @@ void topology::do_sort_by_proximity(locator::host_id address, host_id_vector_rep
         return info{ id, distance(address, loc, id, loc1) };
     }), std::back_inserter(host_infos));
     std::ranges::sort(host_infos, std::ranges::less{}, std::mem_fn(&info::distance));
+    auto it = host_infos.begin();
+    auto prev = it;
+    auto shuffler = _random_engine();
+    for (++it; it < host_infos.end(); ++it) {
+        // Shuffle equal-distance replicas to improve load-balancing
+        if (prev->distance == it->distance) {
+            if (shuffler & 1) {
+                std::swap(prev->id, it->id);
+            }
+            shuffler = std::rotr(shuffler, 1);
+        }
+        prev = it;
+    }
```

Comment:
Idea: instead of a post-sort pass, add random bits to the distance metric. This will cause sort() to randomize the order of otherwise equidistant nodes.

Reply:
I'll try that.  I think that drawing 3 bits (since 64 is not divisible by 3) at a time out of the sampled 64-bit shuffle can do a good job.

Reply:
Tried with v6, and it performs slightly worse than inline shuffle in copy pass.
random-bits-concat:
```
test                                               iterations      median         mad         min         max      allocs       tasks        inst      cycles
    sort_by_proximity_topology.perf_sort_by_proximity    18556938    53.934ns     0.075ns    53.857ns    54.621ns       0.000       0.000       649.8       160.9
```

inline-shuffle:
```
sort_by_proximity_topology.perf_sort_by_proximity    19689561    50.195ns     0.119ns    50.076ns    51.145ns       0.000       0.000       622.5       150.6
```